src/lib/components/quiz/quiz.svelte

```diff
@@ -0,0 +1,249 @@
+<script lang="ts">
+  import type { QuizData, TeamMatch } from '$lib/constants/quiz';
+  import LeftArrow from '$lib/components/icons/left-arrow.svelte';
+  import RightArrow from '$lib/components/icons/right-arrow.svelte';
+  import MoreInfo from './more-info.svelte';
+
+  export let data: QuizData;
+
+  $: index = 0;
+  let response: TeamMatch[] = [];
+  $: answeredAllQuestions =
+    response.length === data.questions.length && !response.includes(undefined);
+  let showResults = false;
+  let showMoreInfo = false;
+  let showTeam: string;
+  $: talliedResponses = response.reduce((tallies, match) => {
+    if (tallies[match]) tallies[match]++;
+    else tallies[match] = 1;
+    return tallies;
+  }, {});
+  let teams = ['Algo', 'Dev', 'Design', 'AI'];
+  let match: string;
+
+  function goLeft() {
+    if (index > 0) {
+      index--;
+    }
+  }
+
+  function goRight() {
+    if (index < data.questions.length - 1) {
+      index++;
+    }
+    if (index === data.questions.length) {
+      // need to display text at the bottom that says:
+      console.log('please answer all questions');
+    }
+  }
+
+  function recordAnswer(match: TeamMatch) {
+    console.log(match);
+
+    response[index] = match;
+    // if answered all questions and no more questions then submitResponses
+    if (!answeredAllQuestions) {
+      goRight();
+    }
```

Comment:
The `goRight` function might contain sufficient checks before going right.
```suggestion
    goRight();
```

Reply:
Done